okhttp/src/nonJvmMain/kotlin/okhttp3/internal/-HostnamesNonJvm.kt

```diff
@@ -17,44 +17,7 @@ package okhttp3.internal
 
 import com.squareup.okhttpicu.SYSTEM_NORMALIZER
 
-internal actual fun String.toCanonicalHost(): String? {
-  val host: String = this
-
-  // If the input contains a :, it’s an IPv6 address.
-  if (":" in host) {
-    // If the input is encased in square braces "[...]", drop 'em.
-    val inetAddressByteArray = (if (host.startsWith("[") && host.endsWith("]")) {
-      decodeIpv6(host, 1, host.length - 1)
-    } else {
-      decodeIpv6(host, 0, host.length)
-    }) ?: return null
-    // TODO implement properly
-    return inet6AddressToAscii(inetAddressByteArray)
-  }
-
-  try {
-    val result = idnToAscii(host)
-    if (result.isEmpty()) return null
-
-    return if (result.containsInvalidHostnameAsciiCodes()) {
-      // The IDN ToASCII result contains illegal characters.
-      null
-    } else if (result.containsInvalidLabelLengths()) {
-      // The IDN ToASCII result contains invalid labels.
-      null
-    } else {
-      result
-    }
-  } catch (_: IllegalArgumentException) {
-    return null
-  }
-}
-
-internal fun inet4AddressToAscii(address: ByteArray): String {
-  return address.joinToString(".")
```

Comment:
I think this might have printed `-1.-1.-1.-1` instead of `255.255.255.255` because we’re using signed bytes. Sigh.

Reply:
In my defense, it doesn't look like it's even called anywhere :)

Reply:
All good. Getting the tests moved over is an important step!